tests/Util/Examples/FlagTest.cs

```diff
@@ -0,0 +1,231 @@
+
+            // After resetting, the flag is empty again.
+            stringFlag.Reset();
+            Assert.AreEqual(default(string), stringFlag.Value);
+        }
+
+        /// <summary>
+        /// Test for <see cref="Flags.Parse(IEnumerable{string}, Flag[])"/> method.
+        /// </summary>
+        [Test]
+        public void TestParse()
+        {
+            Flag<string> stringFlags = new Flag<string>($"--{TestExample.STRING_ARGS_NAME}=",
+                "", true);
+            Flag<long> longFlags = new Flag<long>($"--{TestExample.LONG_ARGS_NAME}=", "", true);
+            Flag<string> singleStringFlag =
+                new Flag<string>($"--{TestExample.SINGLE_STRING_ARG_NAME}=", "", false);
+            Flag<long> singleLongFlag =
+                new Flag<long>($"--{TestExample.SINGLE_LONG_ARG_NAME}=", "", false);
+
+            string[] args = new string[]
+            {
+                $"--{TestExample.STRING_ARGS_NAME}={TestExample.STRING_ARG1}",
+                $"--{TestExample.STRING_ARGS_NAME}={TestExample.STRING_ARG2}",
+                $"--{TestExample.LONG_ARGS_NAME}={TestExample.LONG_ARG1}",
+                $"--{TestExample.LONG_ARGS_NAME}={TestExample.LONG_ARG2}",
+                $"--{TestExample.LONG_ARGS_NAME}={TestExample.LONG_ARG3}",
+                $"--{TestExample.SINGLE_STRING_ARG_NAME}={TestExample.SINGLE_STRING_ARG}",
+                $"--{TestExample.SINGLE_LONG_ARG_NAME}={TestExample.SINGLE_LONG_ARG}"
+            };
+            Flag[] flags = new Flag[]
+            {
+                stringFlags, longFlags, singleStringFlag, singleLongFlag
+            };
+
+            Flags.Parse(args, flags);
+
+            Assert.AreEqual(stringFlags.Values, new string[] {
+                TestExample.STRING_ARG1, TestExample.STRING_ARG2 });
+            Assert.AreEqual(longFlags.Values, new long[] {
+                TestExample.LONG_ARG1, TestExample.LONG_ARG2, TestExample.LONG_ARG3 });
+            Assert.AreEqual(singleStringFlag.Value, TestExample.SINGLE_STRING_ARG);
+            Assert.AreEqual(singleLongFlag.Value, TestExample.SINGLE_LONG_ARG);
+        }
+
+        /// <summary>
+        /// Test for <see cref="Flags.FromParameterInfo(ParameterInfo)"/> method.
+        /// </summary>
+        [Test]
+        public void TestFromParameterInfo()
+        {
+            MethodInfo methodInfo = this.GetType().GetMethod(
+                "TestMethod", BindingFlags.NonPublic | BindingFlags.Instance);
+
+            ParameterInfo[] paramInfos = methodInfo.GetParameters();
+
+            // Convert the first parameter info into flag and verify its properties.
+            Flag<int> firstFlag = Flags.FromParameterInfo(paramInfos[0]) as Flag<int>;
+            Assert.IsNotNull(firstFlag);
+            Assert.IsTrue(firstFlag.IsRepeatable);
+            Assert.AreEqual($"--intArgs=", firstFlag.Prototype);
+
+            // Convert the first parameter info into flag and verify its properties.
```

Comment:
"First" -> "Second".  Also is it worth pulling this logic out to function?  I don't know how re-usable it would be, but a suggestion.